### PR TITLE
Nfsdist tool

### PR DESCRIPTION
**Pull request by samuelnair**
This is a pull request to add the nfsdist tool, which is to compliment the nfsslower tool that i had added in earlier and in keeping with the suite of {etx4,xfs,zfs}(slower|dist) tools that we currently have.

As before, following the CONTRIBUTING-SCRIPTS.md guidelines and added in the smoke test as well.

### CODE REVIEW

**Comment by goldshtn** [comment]
[buildbot, ok to test]

File: tools/nfsdist_example.txt [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+       128 -> 255        : 0        |                                        |
+       256 -> 511        : 2        |****************************************|
+
+
+In this example you can see that the read traffic is rather bi-modal, with about
+26K reads falling within 8 - 30 usecs and about 18K reads spread between 128 -
+8191 usecs. Write traffic is largely clusstered in the 64 - 127 usecs bracket.
+The faster read traffic is probably coming from a filesystem cache and the slower
+traffic from disk. The reason why the writes are so consistenly fast is becuase
+this example test was run on a couple of VM's and i believe the hypervisor was
```
Comment (goldshtn) [inline tools/nfsdist_example.txt]: Capital "I".

File: tools/nfsdist_example.txt [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        64 -> 127        : 0        |                                        |
+       128 -> 255        : 0        |                                        |
+       256 -> 511        : 2        |****************************************|
+
+
+In this example you can see that the read traffic is rather bi-modal, with about
+26K reads falling within 8 - 30 usecs and about 18K reads spread between 128 -
+8191 usecs. Write traffic is largely clusstered in the 64 - 127 usecs bracket.
+The faster read traffic is probably coming from a filesystem cache and the slower
+traffic from disk. The reason why the writes are so consistenly fast is becuase
```
Comment (goldshtn) [inline tools/nfsdist_example.txt]: "because"

File: tools/nfsdist_example.txt [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        16 -> 31         : 0        |                                        |
+        32 -> 63         : 0        |                                        |
+        64 -> 127        : 0        |                                        |
+       128 -> 255        : 0        |                                        |
+       256 -> 511        : 2        |****************************************|
+
+
+In this example you can see that the read traffic is rather bi-modal, with about
+26K reads falling within 8 - 30 usecs and about 18K reads spread between 128 -
+8191 usecs. Write traffic is largely clusstered in the 64 - 127 usecs bracket.
```
Comment (goldshtn) [inline tools/nfsdist_example.txt]: "clustered"

File: tools/nfsdist_example.txt [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+caching all the write traffic to memory.
+
+This "latency" is measured from when the operation was issued from the VFS
+interface to the file system, to when it completed. This spans everything:
+RPC latency, network latency, file system CPU cycles, file system locks, run
+queue latency, etc. This is a better measure of the latency suffered by
+applications reading from a NFS share and can better expose problems
+experienced by NFS clients.
+
+Note that this only traces the common NFS operations (read,write,open and
```
Comment (goldshtn) [inline tools/nfsdist_example.txt]: Spaces between operations: "read, write, open, and ..."

File: tools/nfsdist_example.txt [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+11:02:41:
+
+operation = write
+     msecs               : count     distribution
+         0 -> 1          : 0        |                                        |
+         2 -> 3          : 21       |******                                  |
+         4 -> 7          : 137      |****************************************|
+         8 -> 15         : 3        |                                        |
+
+This shows a write workload, with writes hovering primarily in the 4-7ms range
```
Comment (goldshtn) [inline tools/nfsdist_example.txt]: Period?

**Comment by samuelnair** [comment]
Damn, I should have run it through ispell. Thanks for the spell check @goldshtn

**Comment by goldshtn** [comment]
LGTM! Thanks.

**Pull request merged**